### ISSUE TITLE
Unexpect error when use `default` function in expression of generated column

### ISSUE THREAD

**Issue by Deardrops** [pingcap/tidb#12471]
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do?
If possible, provide a recipe for reproducing the error.
```sql
create table t1(a int default 9, b int as (default(a)));
insert into t1 values(1, default);
select * from t1;
```

2. What did you expect to see?
![image](https://user-images.githubusercontent.com/9463871/65828360-9e62ed80-e2cc-11e9-8e31-d9bb82ca65fe.png)

3. What did you see instead?
![image](https://user-images.githubusercontent.com/9463871/65828370-b63a7180-e2cc-11e9-856f-7440ed2b2046.png)

4. What version of TiDB are you using (`tidb-server -V` or run `select tidb_version();` on TiDB)?
```
Release Version: v4.0.0-alpha-341-g0df9bb7b8-dirty
```

**Comment by Deardrops** [comment]
Related PR: https://github.com/pingcap/tidb/pull/8540

The issue is caused by the check in
https://github.com/pingcap/tidb/blob/41ac571d7b0e333f7b8fc061726b598eddb3825c/planner/core/expression_rewriter.go#L1508-L1514

Same as sql statement `select default(c) from (select (a+1) as c from t) as t0`, When use `default()` function in expression of generated columns, its `colExpr` also has an empty field of `OrigTblName.O`.So the latter will throw an unexpected error.

Any suggessions for fixup this issue?

**Comment by Deardrops** [comment]
Fixed in #12550
